src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -593,17 +617,18 @@ private void GetEvaluationResultsAndContinue(ArrayBuilder<EvalResultDataItem> ro
             else
             {
                 CreateEvaluationResultAndContinue(rows[index], workList, inspectionContext, stackFrame,
-                    result => ContinueWithExceptionHandling(
-                        () => StoreResultAndContinue(result, rows, results, index, numRows, workList, inspectionContext, stackFrame, completionRoutine),
-                        e =>
+                    result => workList.ContinueWith(
+                        () =>
                         {
-                            // If we fail to store a result, just stop enumerating rows (rather than attempting to store
-                            // an error message in the current row, etc).  This is because it may have been the act of
-                            // indexing into the results store that threw (so it would just throw again here).  The user
-                            // experience is less than ideal (there's no real indication that something went wrong),
-                            // however, it seems like it's better to enumerate/display some rows than none.  We will
-                            // receive a non-fatal Watson report in this case, so the problem won't go unnoticed.
-                            completionRoutine();
+                            results[index] = result;
+                            if (index == numRows - 1)
```

Comment:
Nitpick:  I'd personally prefer to switch the if and the else and make the condition (index < numRows).  Even if it can't happen with the current code, we shouldn't call GetEvaluationResultsAndContinue if "index > numRows".